scripts/buildPluginJson.mjs

```diff
@@ -0,0 +1,50 @@
+import YAML from "yaml";
+import fsex from "fs-extra";
+import path from "path";
+import filehound from "filehound";
+import _ from "lodash";
+
+function main() {
+  console.log("Jai Guru");
+  const pluginFiles = filehound
+    .create()
+    .paths("./plugins")
+    .ignoreHiddenDirectories()
+    .ignoreHiddenFiles()
+    .glob(["content.yaml"])
+    .findSync();
+  //console.log(pluginFiles)
+  const pluginsJSON = [];
+  if (pluginFiles && pluginFiles.length > 0) {
+    pluginFiles.forEach((pf) => {
+      const pluginDirs = path.dirname(pf).split(path.sep);
+      const pluginName = pluginDirs[pluginDirs.length - 1];
+      const strYAML = fsex.readFileSync(pf).toString();
+      const doc = YAML.parse(strYAML, "utf-8");
+
+      //find the plugin image name
+      let image;
+      const exampleYAML = doc.example;
+      if (exampleYAML) {
+        const exampleYamlDoc = YAML.parse(exampleYAML);
+        if (exampleYamlDoc) {
+          image = exampleYamlDoc.steps[0].image;
+        }
+      }
```

Comment:
technically the source of truth should be the image value in the main yaml - I appreciate you may be wanting to get rid of the URL beforehand but it might be just better to strip `https://hub.docker.com/r/plugins/` if present

Reply:
its a fair question - long story short, if drone sees a plugin image name like `myPlugin` it assumes the image repo is docker hub - so we can pretty safely strip the docker part

if a plugin lives anywhere but docker hub we need to pass it in as e.g. `quay.io/myPlugin` - so in any other instance we could probably just strip the `https://`

OR we just pass the user the image value (from the yaml) in every instance - then we don't need to worry about edge cases

Reply:
for now let us leave the image extracting logic as is i.e from examples. Since this is a generator that should not harm us much.